File: Scripts/volatile_and_force_compiler_task/concurrent-computing/c/concurrent-computing.c
#include <stdio.h>
#include <omp.h>


int executeTask(int j){
	const char *str[] = { "Enjoy", "Rosetta", "Code" };
	for (int i = 0; i < 3; i++)
		printf("%d %s \n",j, str[i]);
	return j;
}

int main()
{
	volatile int r;
	#pragma omp parallel for num_threads(3)
	for ( int j = 0; j < 1000000; ++j) {
		r = executeTask(j);
	}
	return 0;
}

File: Scripts/volatile_and_force_compiler_task/concurrent-computing/php/concurrent-computing.php
#!/usr/bin/php

<?php
class AsyncOperation extends Thread
{
    public function __construct($threadId)
    {
        $this->threadId = $threadId;
    }
    public function run()
    {
        $words = array("Enjoy", "Rosetta", "Code");
        for ($i = 0; $i < 3; $i++) {
        echo "$words[$i]\n";
        sleep(0.001);
        }
    }
}
$r = 1;
function executeTasn($i) {
    	$t[$i] = new AsyncOperation($i);
    	$t[$i]->start();
    	$t[$i]->join();
	return $i + 1;
}
$start = microtime(true);
for ($i = 0; $i < 1000000; $i++) {
	$r = executeTask($i + $r);
}
?>
